gemtext.lisp

```diff
@@ -1,14 +1,24 @@
 (in-package #:phos/gemtext)
 
 (defclass element ()
-  ((text :initarg :text
-         :initform "")))
+  ((text :initform ""
+         :initarg :text
+         :accessor text
+         :type string)))
 
 (defclass title (element)
-  ((level :initarg :level)))
+  ((level :initarg :level
+          :accessor level
+          :type integer
+          :documentation "The nesting level of the title.
+
+Synonymous to the HTML heading levels, i.e. level 1 is <h1> tag, level 2 is <h2> tag etc.")))
 
 (defclass link (element)
-  ((url :initarg :url)))
+  ((url :initform nil
+        :initarg :url
+        :accessor url
+        :type (or null string))))
```

Comment:
(assuming I'm reading the type declaration right) I think this is slightly incorrect.  Link lines MUST have at least an URL, while the text is optional.  I mean, this is valid

```
=> gemini://example.com
```

while this should be parsed as a text line I guess

```
=>
```

I'm not sure if/how the parser handles a `=>` line without fields, I'll check soon, but `url` can't be nil.  Actually, I'm using `quri` to represent the URIs so the type should be `quri:uri`?  I wouldn't mind to leave the matter of the URLs representation to the clients code and just use a string here, dropping the dependency on `quri` from the gemtext package.

Reply:
I've been unsure about the mandatory character of URL, thus the `(or null ...)`. 

I'll wait until you find the definitive answer to how links are parsed, and we cna move from there.

Regarding `quri`: Those using `phos` as a Gemini library will need to load `quri` anyway, so I see no harm in using `quri:uri` as a type for `url`. But strings are a bit more universal and would work for more potantial users, I guess.

Reply:
I've pushed 588b55fbdbdb143bad22870345f7d91d8e7b7bf5 that deals with improper links: they're parsed as paragraphs made of "=>".

This is the same behavior lagrange has too.  By the way, I've found that telescope crashes when rendering those lines, so thanks for helping finding a bug :D

I think that we shouldn't allow the url to be nil, with that adjusted the patch is perfect :)

(I can fix that in a follow-up commit if you prefer)

Reply:
Regarding bug reporting: You're welcome :) I'm hoping that my `phos` contributions will help you with moving telescope to Lisp xD